### PR TITLE
#282: Fix sourcetype IN not working

### PR DESCRIPTION
**Pull request by eemhu**
also cleanup searchQualifier in LogicalStatementCatalyst/XML, enable sourcetype IN test in logicalOperationTest

Test note:
Assertions were changed as they were incorrect for the data used

### CODE REVIEW

File: src/main/java/com/teragrep/pth10/ast/commands/logicalstatement/LogicalStatementCatalyst.java [hunk 1]
```diff
@@ -478,6 +472,17 @@ else if (left.getSymbol().getType() == DPLLexer.INDEX_IN) {
 
             }
         }
+        else if (left.getSymbol().getType() == DPLLexer.SOURCETYPE && operation.getSymbol().getType() == DPLLexer.IN) {
+            for (DPLParser.StringTypeContext stringType : ctx.stringType()) {
+                final String sourceType = new UnquotedText(new TextString(stringType.getText())).read();
+                final String rlikeStatement = glob2rlike(sourceType);
+                if (sQualifier == null) {
```
Comment (51-code) [inline src/main/java/com/teragrep/pth10/ast/commands/logicalstatement/LogicalStatementCatalyst.java:479]: Is null needed here? Working with Spark columns is quite difficult without null and the for loops kind of require the check. But could it be possible to initialize the first `sQualifier` outside the for loop so null is not needed? Or maybe new objects.

Comment (eemhu) [reply src/main/java/com/teragrep/pth10/ast/commands/logicalstatement/LogicalStatementCatalyst.java]: added OrColumn object

File: src/main/java/com/teragrep/pth10/ast/commands/logicalstatement/LogicalStatementXML.java
Comment (kortemik) [inline src/main/java/com/teragrep/pth10/ast/commands/logicalstatement/LogicalStatementXML.java:1]: no tests regarding xml generation, please create them for the IN statement

File: src/main/java/com/teragrep/pth10/ast/commands/logicalstatement/LogicalStatementCatalyst.java
Comment (kortemik) [inline src/main/java/com/teragrep/pth10/ast/commands/logicalstatement/LogicalStatementCatalyst.java:1]: please create translation tests for IN statement from DPL to Spark SQL

**Review by kortemik — CHANGES_REQUESTED** [review]

File: src/main/java/com/teragrep/pth10/ast/commands/logicalstatement/LogicalStatementXML.java
Comment (eemhu) [reply src/main/java/com/teragrep/pth10/ast/commands/logicalstatement/LogicalStatementXML.java:1]: added

**Review by kortemik — APPROVED** [review]

**Comment by Abigael-JT** [comment]
fixes #282

**Pull request merged**